### PR TITLE
Update RadMon, add support for orion, completes #14

### PR DESCRIPTION
**Pull request by EdwardSafford-NOAA**
This PR includes:

- Updates to non-operational portions of RadMon (primarily plotting functionality) to work correctly in the GSI-monitor repo.  
- Added run-time lua modules to load necessary modules from hpc stack for non-operational portions of RadMon (i.e. plotting, off-line data extraction, utility functions).  If there's a different/more preferred method to handle this I'll be glad to make any requested changes.
- Added a GSI-monitor/parm/Mon_config file.  This is intended to contain all configuration info and settings common to all 4 monitors.   
- Added support for orion, removed all references to wcoss_dell and wcoss_cray.
- Fixed two bugs.

### CODE REVIEW

**Comment by aerorahul** [comment]
Adding @WalterKolczynski-NOAA to review for an extra pair of eyes.

File: parm/Mon_config [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+export MON_PTMP=${MON_PTMP:-$ptmp}
+export MON_STMP=${MON_STMP:-$stmp}
+export MON_LOGDIR=${MON_LOGDIR:-$ptmp/logs}
+export TANKDIR=${TANKDIR:-$tankdir}
+
+export WEBUSER=${WEBUSER:-$USER}
+export WEBSVR=${WEBSVR:-emcrzdm}
+
+
+export Z=${Z:-"gz"}
```
Comment (aerorahul) [inline parm/Mon_config:117]: ooh. single character variables are usually frowned upon unless they represent an index e.g. `i, j, k` or a dimension `x, y, z`.

File: src/Radiance_Monitor/nwprod/gdas_radmon/driver/test_jgdas_verfrad_hera.sh [hunk 2]
```diff
@@ -40,27 +40,30 @@ export global_shared_ver=v14.1.0
 #############################################################
 
 export RADMON_SUFFIX=${RADMON_SUFFIX:-testrad}
-export NWTEST=${NWTEST:-/scratch1/NCEPDEV/da/Edward.Safford/noscrub/GSI/util/Radiance_Monitor/nwprod}
+export NWTEST=${NWTEST:-/scratch1/NCEPDEV/da/Edward.Safford/noscrub/GSI-monitor/src/Radiance_Monitor/nwprod}
```
Comment (aerorahul) [inline src/Radiance_Monitor/nwprod/gdas_radmon/driver/test_jgdas_verfrad_hera.sh:43]: Is it too hard-wired to replace with `$USER`.  It should have the desired effect.

File: src/Radiance_Monitor/nwprod/gdas_radmon/driver/test_jgdas_verfrad_orion.sh [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+#SBATCH -J gdas_verfrad
+#SBATCH --ntasks=1 --mem=5g
+#SBATCH --time=20
+#SBATCH --account=da-cpu
+#SBATCH -D .
+
+
+set -x
+
+export MY_MACHINE=hera
```
Comment (aerorahul) [inline src/Radiance_Monitor/nwprod/gdas_radmon/driver/test_jgdas_verfrad_orion.sh]: The script is for orion, while this variable says hera.

**Review by aerorahul — APPROVED** [review]
I don't think I have enough knowledge to comment on the changes in this PR.
I did look through the changes, and just a few curious questions:
- adding the `run` versions of the modulefiles is interesting.
- see other comments in line where I found curious items.
- There is a lot of repetition in these scripts, but I suppose thats how it has been.

File: src/Radiance_Monitor/nwprod/gdas_radmon/driver/test_jgdas_verfrad_orion.sh (hunk 3 above)
Comment (EdwardSafford-NOAA) [reply src/Radiance_Monitor/nwprod/gdas_radmon/driver/test_jgdas_verfrad_orion.sh]: Oh my.  And that it runs is a good indication it serves no purpose.  I'll remove it.

File: src/Radiance_Monitor/nwprod/gdas_radmon/driver/test_jgdas_verfrad_hera.sh (hunk 2 above)
Comment (EdwardSafford-NOAA) [reply src/Radiance_Monitor/nwprod/gdas_radmon/driver/test_jgdas_verfrad_hera.sh:43]: Can do.

File: parm/Mon_config (hunk 1 above)
Comment (EdwardSafford-NOAA) [reply parm/Mon_config:117]: Fair point.  I'd like to fix that as a separate issue if you're OK with that.

**Comment by EdwardSafford-NOAA** [comment]
@aerorahul as to the repetition, I agree.  The basic design itself was never very concise.  In these, and in  future changes, I'm very deliberately trying to strike a balance between making it work well now without rewriting large portions of the package.  As we've discussed there are plans to sun-set this package and replace it with a more robust solution that follows modern programming standards.

I'll add David Huber to the review list -- he has experience with the monitors; he has ported them to s4 and Jet, and he can comment directly on the functioning.  I wanted your input exactly as you gave it -- to make sure I was in accord with the general direction things should be moving in, and I really wanted your opinion on the run-time lua mods.  Thanks.

File: modulefiles/common-run.lua [hunk 4]
```diff
@@ -0,0 +1,10 @@
+help([[
+Load common modules to run GSI Monitoring on all machines
+]])
+
+local wgrib2=os.getenv("wgrib2") or "2.0.8"
+local prod_util=os.getenv("prod_util") or "1.2.2"
```
Comment (DavidHuber-NOAA) [inline modulefiles/common-run.lua]: These should be wgrib2_ver and prod_util_ver.

File: modulefiles/hera.gnu-run.lua [hunk 5]
```diff
@@ -0,0 +1,16 @@
+help([[
+]])
+
+prepend_path("MODULEPATH", "/scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/modulefiles/stack")
+
+local hpc_ver=os.getenv("hpc_ver") or "1.1.0"
+local hpc_gnu_ver=os.getenv("hpc_gnu_ver") or "9.2.0"
+local grads=os.getenv("grads") or "2.2.1"
```
Comment (DavidHuber-NOAA) [inline modulefiles/hera.gnu-run.lua]: Should be grads_ver.

File: modulefiles/hera.intel-run.lua [hunk 6]
```diff
@@ -0,0 +1,17 @@
+help([[
+]])
+
+prepend_path("MODULEPATH", "/scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/modulefiles/stack")
+
+local hpc_ver=os.getenv("hpc_ver") or "1.1.0"
+local hpc_intel_ver=os.getenv("hpc_intel_ver") or "18.0.5.274"
+local grads=os.getenv("grads") or "2.2.1"
```
Comment (DavidHuber-NOAA) [inline modulefiles/hera.intel-run.lua]: grads_ver

File: modulefiles/orion-run.lua [hunk 7]
```diff
@@ -0,0 +1,16 @@
+help([[
+]])
+
+prepend_path("MODULEPATH", "/apps/contrib/NCEP/libs/hpc-stack/modulefiles/stack")
+
+local hpc_ver=os.getenv("hpc_ver") or "1.1.0"
+local hpc_intel_ver=os.getenv("hpc_intel_ver") or "2018.4"
+local grads=os.getenv("grads") or "2.2.1"
```
Comment (DavidHuber-NOAA) [inline modulefiles/orion-run.lua]: grads_ver

File: modulefiles/wcoss2-run.lua [hunk 8]
```diff
@@ -0,0 +1,16 @@
+help([[
+]])
+
+local intel_ver=os.getenv("intel_ver") or "19.1.3.304"
+
+load(pathJoin("intel", intel_ver))
+
+
+prepend_path("MODULEPATH", "/apps/test/lmodules/core/")
+load ("GrADS/2.2.2")
```
Comment (DavidHuber-NOAA) [inline modulefiles/wcoss2-run.lua:10]: Should grads_ver be used here to follow other templates?

**Review by DavidHuber-NOAA — CHANGES_REQUESTED** [review]
Just a few minor changes to the module files.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

File: modulefiles/common-run.lua (hunk 4 above)
Comment (EdwardSafford-NOAA) [reply modulefiles/common-run.lua]: Yup, fixed.

File: modulefiles/hera.intel-run.lua (hunk 6 above)
Comment (EdwardSafford-NOAA) [reply modulefiles/hera.intel-run.lua]: yup, fixed.

File: modulefiles/wcoss2-run.lua (hunk 8 above)
Comment (EdwardSafford-NOAA) [reply modulefiles/wcoss2-run.lua:10]: Not yet.  GrADS isn't yet in the stack on wcoss2.  When it does arrive I figure to move the grads load to common-run.lua.

**Pull request merged**